### PR TITLE
Ex 1

### PR DESCRIPTION
**Pull request by tvlinh0com**
Anh review cho em cái bài tập này

### CODE REVIEW

File: ex-1/define.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
+
+
+#define STRING_LENGTH 300
+#define STRING_WORDS_NUMBER 30
+#define WORD_LENGTH 10
+
+
+int sumOdd(unsigned int);
```
Comment (nguyenvangiang1988) [inline ex-1/define.h]: Please make the function names consistently start with upper case letter. For ex: SumOdd.

File: ex-1/entry.c [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+void PrintLine(char* s){
+	printf("%s\n", s);
+}
+
+void Print(char* s){
+	printf("%s", s);
+}
+
+int main(){
+	PrintLine("Choose Problem");
```
Comment (nguyenvangiang1988) [inline ex-1/entry.c]: Please use 3 or 4 spaces character instead of using "tab"

File: ex-1/entry.c [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	PrintLine("1. Sum Odd");
+	PrintLine("2. Word Swap");
+	PrintLine("3. Check Student Attendance");
+
+
+	int probid;
+	Print("Problem number: ");
+	scanf("%d", &probid);
+
+	if(probid==1){
```
Comment (nguyenvangiang1988) [inline ex-1/entry.c]: Let's have a space between if and "("

File: ex-1/findabsence.c [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+void DumpArray(int* a, int n){
+	for(int i=0; i<n; i++){
+		printf("%d ", a[i]);
+	}
+	printf("\012");
+}
+
+
+int inarray(int e, int* a, int n){
```
Comment (nguyenvangiang1988) [inline ex-1/findabsence.c]: PrintArray ???

File: ex-1/findabsence.c [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+	for(int i = 0; i < n; i++){
```
Comment (nguyenvangiang1988) [inline ex-1/findabsence.c]: Need a space between for and "("

File: ex-1/findabsence.c [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+	for(int i = 1; i <= length; i++){
+		if(!inarray(i, ids, length)){
+			result[nabsence++] = i;
+		}
+	}
+
+//	Sort(result, nabsence);
+	DumpArray(result, nabsence);
+}
```
Comment (nguyenvangiang1988) [inline ex-1/findabsence.c]: Need to place a new line at the end of file.

File: ex-1/define.h (hunk 1 above)
Comment (tvlinh0com) [reply ex-1/define.h]: OK anh.

File: ex-1/wordswap.c [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+	}
+
+	int length = strlen(str);
+	char* res = (char*)malloc(length + 1);
+	strcpy(res, "");
+
+	int fsi, fei, lsi, lei;
+
+	fsi = indexofnonspace(str, indexofspace(str, indexofnonspace(str, 0, 1), 1), 1);
+//	fsi = indexofspace(str, indexofnonspace(str, 0, 1), 1) + 1;
```
Comment (nguyenvangiang1988) [inline ex-1/wordswap.c]: Next time please remove all dead codes.

**Pull request merged**